init.lua

```diff
@@ -719,20 +719,38 @@ function PaperWM:cycleWindowSize(direction)
         return new_size
     end
 
+    local function findNextSizeDescending(area_size, frame_size)
```

Comment:
This function is very similar to findNextSizeAscending (aka findNewSize). Maybe instead of a separate findNextSizeDecending function, you could pass the cycle_direction parameter into findNewSize and either iterate up the sizes table or down the sizes table based on cycle_direction.